app/lib/core/view_models/add_history_view_model.dart

```diff
@@ -0,0 +1,45 @@
+import 'package:app/core/models/test_history.dart';
+import 'package:app/core/services/firestore/firestore_service.dart';
+import 'package:app/core/services/navigation/nav_service.dart';
+import 'package:app/core/services/service_locator.dart';
+import 'package:app/core/view_models/base_view_model.dart';
+import 'package:flutter/material.dart';
+
+class AddHistoryViewModel extends BaseViewModel {
+  final FirestoreService _firestoreService = locator<FirestoreService>();
+  final NavigationService _navigationService = locator<NavigationService>();
+
+  HistoryRecord _edittingHistory;
+  bool get _isEditting => _edittingHistory != null;
+
+  Future addHistory({@required String title}) async {
+    setBusy(true);
+    var result;
+
+    if (!_isEditting) {
+      result = await _firestoreService.addHistory(HistoryRecord(
+        title: title,
+        userId: currentUser.userId,
+      ));
+    } else {
+      result = await _firestoreService.updatHistoryTravel(HistoryRecord(
+          title: title,
+          userId: _edittingHistory.userId,
+          documentId: _edittingHistory.documentId));
+    }
+
+    setBusy(false);
+
+    // If result is type String, then something went wrong
```

Comment:
Our new error types you will add ( :) ) should make this better!

Reply:
lol

Reply:
Well, you're checking if the result is a `String` type, which is a smelly way of checking if something went wrong. You should either:
- [ ] Check the contents of the string for a notable reason that the result is a failure, or
- [ ] Use the `Response` class I suggested to make this more understandable:

```
var Response response;
// set response
response.isFailure() ? print('Something went wrong: $response.message') : print('Successfully added');

Reply:
Or even better:
```
response.ifFailure((res) => print('Had a problem adding history: ${res.error}')
                .ifSuccess((res) => print('Successfully added history');

Reply:
The code highlighted to your review is `setBusy(false)`. So, I thought you were referring to that code. In my recent change, I did not check if the result is in type String. Instead, I checked if the response from ResultStatus is successful or now. 
```
 if (result == ResultStatus.successful) {
      await _dialogService.showDialog(
        title: 'Travel History',
        description: 'Successfully added!',
      );
    } else {
      final errorMessage = ExceptionHandler.exceptionMessage(result);
      await _dialogService.showDialog(
        title: 'Travel History Creation Failed',
        description: errorMessage,
      );
    }
```
I can simply use a shorthand if-else  here, but I honestly think this is readable and understandable this way

Reply:
Oh, the code for the review comment is the line immediately before the comment:
![Screen Shot 2021-02-05 at 8 13 59 PM](https://user-images.githubusercontent.com/32527219/107105962-b0bd3780-67ee-11eb-93f6-c86d488414df.png)